### PR TITLE
Add cursor shape support

### LINKED ISSUE #133: [BUG] cosmic-term doesn't respect DECSCUSR change cursor shape
There are escape codes to change cursor shape. These are often used in TUI apps. They currently don't work.

For example, set cursor to bar

```sh
echo -e "\e[5 q"
```
In other terminals (Gnome, kitty, xterm) running these commands makes the cursor turn into a bar.

### PR DESCRIPTION
**Pull request by A-Walrus**
Fixes #133, #175, and #297

[Screencast from 2024-08-23 15-03-47.webm](https://github.com/user-attachments/assets/c0d4a084-1d39-4f2e-b83b-b3c2883400e3)

### CODE REVIEW

File: src/terminal_box.rs [hunk 1]
```diff
@@ -587,6 +588,46 @@ where
...
+                                    (col as f32 * width).floor(),
+                                    ((line + 1) as f32 * height).floor(),
+                                ),
+                            Size::new(width, 1.0),
+                        ),
+                        ..Default::default()
+                    };
+                    renderer.fill_quad(quad, color);
+                }
+                CursorShape::HollowBlock => {} // TODO not sure when this would even be activated
```
Comment (A-Walrus) [inline src/terminal_box.rs:626]: It's not one of the options in `DECSCUSR` and I couldn't find much info about it...
https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h4-Functions-using-CSI-_-ordered-by-the-final-character-lparen-s-rparen:CSI-Ps-SP-q.1D81

Comment (leb-kuchen) [reply src/terminal_box.rs:626]:
```rs
pub enum CursorShape {
    /// Cursor is a block like `▒`.
    #[default]
    Block,

    /// Cursor is an underscore like `_`.
    Underline,

    /// Cursor is a vertical bar `⎸`.
    Beam,

    /// Cursor is a box like `☐`.
    HollowBlock,

    /// Invisible cursor.
    Hidden,
}
```

Comment (A-Walrus) [reply src/terminal_box.rs:626]: Sure, what I'm not sure about is in what case HollowBlock would be selected.

Comment (leb-kuchen) [reply src/terminal_box.rs:626]: It the application overrides the default cursor.

Comment (MoSal) [reply src/terminal_box.rs:626]: @A-Walrus 

You are write to note that no sequence sets this shape.

It's used by alacritty (the app) for unfocused terminals from the looks of it:
https://github.com/alacritty/alacritty/blob/5060f8eeb864e8c304fbad9588bdd882db942356/alacritty/src/display/content.rs#L57-L58

Comment (leb-kuchen) [reply src/terminal_box.rs:626]: I guess eventually there will a config option for setting the cursor shape, though I am not sure whether the UX team likes hollow cursor.

Comment (jackpot51) [reply src/terminal_box.rs:626]: I think it is fine to have HollowBlock not handled yet. The ability to handle underline and beam cursors is already an improvement so I will merge.

**Review by jackpot51 — APPROVED** [review]

**Comment by snaggen** [comment]
I see the comment about the HollowBlock, that is used instead of the block cursor when the terminal doesn't have focus.

**Pull request merged**